solidity/dashboard/src/sagas/messages.js

```diff
@@ -5,13 +5,33 @@ import {
   REMOVE_MESSAGE,
   ADD_MESSAGE,
 } from "../actions/messages"
+import { messageType } from "../components/Message"
 
 function* addMessage(action) {
   yield put({ type: ADD_MESSAGE, payload: action.payload })
 }
 
 function* removeMessage(action) {
-  yield put({ type: REMOVE_MESSAGE, payload: action.payload })
+  yield put({ type: REMOVE_MESSAGE, payload: action.payload.id })
+  if (action.messageType) {
+    // switch for future notification data clearance for other message types
+    switch (action.payload.messageType) {
```

Comment:
I think there is a mismatch with actions. With `switch` statement the `removeMessage` action creator is responsible for 2 things which in my opinion is not a good idea. We can handle it in the `LiquidityRewardsEarnedMessage` by dispatching  `notifications_data/liquidityRewardNotification/pairs_displayed_updated` in `onMessageClose` function.

In `LiquidityRewardsEarnedMessage.jsx`

```
<Message
 ...
 onMessageClose={(messageId) => {
  dispatch({
          type:
            "notifications_data/liquidityRewardNotification/pairs_displayed_updated",
          payload: [],
    })
  props.onMessageClose(messageId)
 }}
/>
```